client/src/pages/LogIn_SignUp.js

```diff
@@ -0,0 +1,208 @@
+import {
+  TextField,
+  Button,
+  Typography,
+  Link,
+  Paper,
+  Grid,
+  Box,
+  makeStyles,
+} from '@material-ui/core';
+import React, { useEffect, useState, useCallback } from 'react';
+import { useLocation } from 'react-router-dom';
+
+const useStyles = makeStyles({
+  textField: {
+    width: '50%',
+  },
+  bold: {
+    color: 'black',
+  },
+  seperator: {
+    width: '10%',
+    border: '1px solid #00e676',
+  },
+  formPadding: {
+    padding: '20px',
+  },
+});
+
+const LogIn_SignUp = () => {
+  //state declaration
+  const [signInEmail, setEmail] = useState('');
+  const [signInPassword, setPassword] = useState('');
+  const [registerName, setName] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [lengthValidation, setLengthValidation] = useState(true);
+  const [confirmValidation, setConfirmValidation] = useState(true);
+
+  const classes = useStyles();
+
+  //Determines which form elements to render based on the path
+  const signInForm = useLocation().pathname === '/login' ? true : false;
+
+  //change this to change the min length requirement
+  const minLength = 6;
+
+  //front-end validation for registering users
+  const handleValidation = useCallback(() => {
+    if (signInPassword !== '' && signInPassword.length < minLength) {
+      setLengthValidation(false);
+    } else {
+      setLengthValidation(true);
+    }
+    if (confirmPassword !== '' && signInPassword !== confirmPassword) {
+      setConfirmValidation(false);
+    } else {
+      setConfirmValidation(true);
+    }
+  }, [signInPassword, confirmPassword]);
+
+  useEffect(() => {
+    handleValidation();
+  }, [signInPassword, confirmPassword, handleValidation]);
+
+  const handleChange = (event) => {
+    const { value, name } = event.target;
+    if (name === 'signInEmail') {
+      setEmail(value);
+      console.log(signInEmail);
+    } else if (name === 'signInPassword') {
+      setPassword(value);
+    } else if (name === 'registerName') {
+      setName(value);
+    } else if (name === 'confirmPassword') {
+      setConfirmPassword(value);
+    }
```

Comment:
You can do this without using a series of if statements by making use of bracket notation to access object properties. That should simplify your logic for storing, accessing, and changing user data

Reply:
made state into an object to simplify the logic